File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.py
class Solution:

    def sumOddLengthSubarrays(self, arr: list[int]) -> int:
        total = 0

        for i in range(len(arr)):
            current_sum = 0

            for j in range(i, len(arr)):
                current_sum += arr[j]

                if (j - i + 1) % 2 != 0:
                    total += current_sum

        return total

File: 1588-sum-of-all-odd-length-subarrays/1588-sum-of-all-odd-length-subarrays.java
class Solution {
    public int sumOddLengthSubarrays(int[] arr) {

        int sum = 0;

        for (int i = 0; i < arr.length; i++) {

            int currentSum = 0;

            for (int j = i; j < arr.length; j++) {

                currentSum += arr[j];

                if ((j - i + 1) % 2 != 0) {
                    sum += currentSum;
                }
            }
        }

        return sum;
    }
}